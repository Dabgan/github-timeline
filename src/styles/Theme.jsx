import React from "react";
import { ThemeProvider } from "styled-components";

const breakpoints = {
    bigDesktop: 1440,
    desktop: 1150,
    bigTablet: 1020,
    tablet: 767,
    phone: 374,
    smallPhone: 340,
};
const device = {
    bigDesktop: `(min-width: ${breakpoints.bigDesktop})`,
    desktop: `(min-width: ${breakpoints.desktop})`,
    bigTablet: `(min-width: ${breakpoints.bigTablet})`,
    tablet: `(min-width: ${breakpoints.tablet})`,
    phone: `(min-width: ${breakpoints.phone})`,
    smallPhone: `(min-width: ${breakpoints.smallPhone})`,
};

const font = {
    family: ["sans-serif", "Roboto"],
    color: { main: "#FCC926", secondary: `#272727`, black: "#000000" },
    weight: {
        regular: "400",
        semibold: "500",
        bold: "700",
    },
    size: {
        header: {
            s: "2.7rem",
            m: "3.6rem",
        },
        body: {
            xs: "1rem",
            s: "1.22rem",
            m: "1.44rem",
        },
    },
};

const colors = {
    main: "#FCC926",
    secondary: "#0DFDC9",
    ternary: "#EF26FC",
    powderWhite: "#FFFDF9",
    persianGreen: "#06B49A",
    lightBlue: "#0DFDC9",
    onyx: "#36313D",
    black: "#000000",
    lightBrown: "#B08600",
};

const theme = {
    device,
    font,
    colors,
    padding: {
        xs: "0.7rem",
        s: "1.5rem",
        m: "2.5rem",
        l: "4rem",
    },
    margin: {
        xs: "0.7rem",
        s: "2rem",
        m: "4rem",
        l: "8rem",
        xl: "13rem",
    },
    borderRadius: "10px",
};

const Theme = ({ children }) => (
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
