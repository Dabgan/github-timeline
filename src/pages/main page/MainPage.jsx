import React, { useState, useEffect } from "react";
import Button from "../../components/Button";
import * as styled from "./mainPage.styles";
import { useHistory } from "react-router-dom";
import { useSelector, useDispatch } from "react-redux";
import { setUsername, updateCommits } from "../../redux";
import axios from "axios";
import formatDate from "../../formatDate";
// import useGetGitHubApiData from "./../../hooks/useGetGitHubApiData";

const MainPage = () => {
    // const username = useSelector((state) => state.username.username);
    const [name, setName] = useState("dabgan");
    const dispatch = useDispatch();
    // useGetGitHubApiData();

    const getGithubApiCommits = (username) => {
        const config = {
            method: "get",
            // url: `https://api.github.com/repos/${username}/Snowshop/commits?page=1`,
            url: `https://api.github.com/users/${username}`,
        };

        axios(config)
            .then((res) => {
                let resCommits = [];
                console.log(res.data);
                const repos = res.data.repos_url;
                console.log(repos);
                return repos;
                // get and logout every commit on page
                // const links = parse(res.headers.link);
                // for (let i = 0; i < res.data.length; i++) {
                //     const date = new Date(res.data[i].commit.committer.date);
                //     resCommits.push({
                //         message: res.data[i].commit.message,
                //         date: formatDate(date),
                //     });
                // }
                // dispatch(updateCommits(resCommits));
                // go to next page and do the same
                // axios(links.next)
                //     .then((res) => {
                //         for (let i = 0; i < res.data.length; i++) {
                //             // console.log(res.data[i].commit.message);
                //             resCommits.push(res.data[i].commit.message);
                //         }
                //         console.log(resCommits);
                //     })
                //     .catch(() => console.log(`there is no next page`));
            })
            .then((res) => {
                axios(res).then((res) => {
                    console.log(res.data);
                });
            })
            .catch((error) => console.log(error));
    };

    let history = useHistory();

    const handleSubmit = (e) => {
        e.preventDefault();
        dispatch(setUsername(name));
        getGithubApiCommits(name);
        history.push("/timeline", {});
    };

    return (
        <styled.MainContainer onSubmit={(e) => handleSubmit(e)}>
            <styled.Title>Github timeline</styled.Title>
            <styled.Container>
                <styled.Label htmlFor="username">Github username</styled.Label>
                <styled.Input
                    required
                    id="username"
                    name="username"
                    type="text"
                    value={name}
                    onChange={(e) => setName(e.target.value)}
                />
            </styled.Container>
            <Button submit>Show commits</Button>
        </styled.MainContainer>
    );
};

export default MainPage;
