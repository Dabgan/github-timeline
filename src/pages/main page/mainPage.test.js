import React from "react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import MainPage from "./MainPage";
import Theme from "../../styles/Theme";

const mockStore = configureStore();

describe("Render main page without errors", () => {
    let store;
    let wrapper;

    beforeAll(() => {
        store = mockStore({
            repos: {
                error: "",
            },
        });
        wrapper = global
            .shallow(
                <Provider store={store}>
                    <Theme>
                        <MainPage />
                    </Theme>
                </Provider>
            )
            .render();
    });

    it("Renders without crashing", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
