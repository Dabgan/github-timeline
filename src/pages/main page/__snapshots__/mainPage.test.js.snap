// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render main page without errors Renders without crashing 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-AxjAm kXPA-Dk",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "sc-AxiKw hXAhKp",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "sc-AxirZ jSoSfK",
            },
            "children": Array [
              Object {
                "data": "Github timeline",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "h1",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "sc-AxhCb fKucvK",
                "for": "username",
              },
              "children": Array [
                Object {
                  "data": "Enter github username:",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "label",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "sc-AxhUy bmauUR",
                  "id": "username",
                  "name": "username",
                  "required": "",
                  "type": "text",
                  "value": "gabriel",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "sc-AxheI gUfVHM",
                    "type": "submit",
                  },
                  "children": Array [
                    Object {
                      "data": "Show commits",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                  "name": undefined,
                  "required": undefined,
                  "type": undefined,
                  "value": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                  "name": undefined,
                  "required": undefined,
                  "type": undefined,
                  "value": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "for": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "for": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "sc-AxhCb fKucvK",
              "for": "username",
            },
            "children": Array [
              Object {
                "data": "Enter github username:",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "label",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "sc-AxhUy bmauUR",
                "id": "username",
                "name": "username",
                "required": "",
                "type": "text",
                "value": "gabriel",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "sc-AxheI gUfVHM",
                  "type": "submit",
                },
                "children": Array [
                  Object {
                    "data": "Show commits",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "type": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "type": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
                "name": undefined,
                "required": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
                "name": undefined,
                "required": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "sc-AxirZ jSoSfK",
              },
              "children": Array [
                Object {
                  "data": "Github timeline",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "h1",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "for": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "for": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "sc-AxhUy bmauUR",
              "id": "username",
              "name": "username",
              "required": "",
              "type": "text",
              "value": "gabriel",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "sc-AxheI gUfVHM",
                "type": "submit",
              },
              "children": Array [
                Object {
                  "data": "Show commits",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "type": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "type": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "sc-AxhCb fKucvK",
                "for": "username",
              },
              "children": Array [
                Object {
                  "data": "Enter github username:",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "label",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "sc-AxirZ jSoSfK",
                },
                "children": Array [
                  Object {
                    "data": "Github timeline",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "h1",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "for": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "for": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "id": undefined,
              "name": undefined,
              "required": undefined,
              "type": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "id": undefined,
              "name": undefined,
              "required": undefined,
              "type": undefined,
              "value": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "sc-AxheI gUfVHM",
              "type": "submit",
            },
            "children": Array [
              Object {
                "data": "Show commits",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "sc-AxhUy bmauUR",
                "id": "username",
                "name": "username",
                "required": "",
                "type": "text",
                "value": "gabriel",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "sc-AxhCb fKucvK",
                  "for": "username",
                },
                "children": Array [
                  Object {
                    "data": "Enter github username:",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "label",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "sc-AxirZ jSoSfK",
                  },
                  "children": Array [
                    Object {
                      "data": "Github timeline",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "h1",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "for": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "for": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "id": undefined,
                "name": undefined,
                "required": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "id": undefined,
                "name": undefined,
                "required": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "type": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "type": undefined,
            },
          },
        ],
        "name": "form",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
